src/compiler/types.rs

```diff
@@ -3157,7 +3239,7 @@ impl DiagnosticRelatedInformationInterface for DiagnosticRelatedInformation {
 #[derive(Clone, Debug)]
 pub struct BaseDiagnosticRelatedInformation {
     code: u32,
-    file: Option<Rc<SourceFile>>,
+    file: Option<Weak<SourceFile>>,
```

Comment:
Ran into some circular debug-logging with this as a strong reference (specifically because `SourceFile` now has `parse_diagnostics`, which here circularly referred back to the `SourceFile`)